packages/calling-component-bindings/src/hooks/useSelector.ts

```diff
@@ -2,15 +2,18 @@
 // Licensed under the MIT license.
 
 import { CallClientState, StatefulCallClient } from 'calling-stateful-client';
-import { useCall, useCallClient } from '../providers';
+import { CallClientContext, useCall } from '../providers';
 
-import { useState, useEffect, useRef, useMemo } from 'react';
+import { useState, useEffect, useRef, useMemo, useContext } from 'react';
 
-export const useSelector = <SelectorT extends (state: CallClientState, props: any) => any>(
-  selector: SelectorT,
+export const useSelector = <
+  SelectorT extends (state: CallClientState, props: any) => any,
+  ParamT extends SelectorT | undefined
+>(
+  selector: ParamT,
   selectorProps?: Parameters<SelectorT>[1]
-): ReturnType<SelectorT> => {
-  const callClient: StatefulCallClient = useCallClient() as any;
+): ParamT extends SelectorT ? ReturnType<SelectorT> : undefined => {
+  const callClient: StatefulCallClient | undefined = useContext(CallClientContext)?.callClient;
```

Comment:
Why can we no longer use `useCallClient` here?

Reply:
A good question! Because useCallClient has a detection of whether callClient is undefined and throw error(that is a designed behavior for api exposed externally), but this place we need to detect whether we are under the call context or not, so this undefine return is necessary